internal/host/plugin/repository_host_catalog_test.go

```diff
@@ -366,12 +367,645 @@ func TestRepository_CreateCatalog(t *testing.T) {
 	})
 }
 
-func assertPluginBasedPublicId(t *testing.T, prefix, actual string) {
-	t.Helper()
-	assert.NotEmpty(t, actual)
-	parts := strings.Split(actual, "_")
-	assert.Equalf(t, 2, len(parts), "want 1 '_' in PublicId, got %d in %q", len(parts), actual)
-	assert.Equalf(t, prefix, parts[0], "PublicId want prefix: %q, got: %q in %q", prefix, parts[0], actual)
+func TestRepository_UpdateCatalog(t *testing.T) {
+	ctx := context.Background()
+	dbConn, _ := db.TestSetup(t, "postgres")
+	dbRW := db.New(dbConn)
+	dbWrapper := db.TestWrapper(t)
+	dbKmsCache := kms.TestKms(t, dbConn, dbWrapper)
+	orgScope, projectScope := iam.TestScopes(t, iam.TestRepo(t, dbConn, dbWrapper))
+
+	// Plugins, OnUpdateCatalogRequest state, and returned error -
+	// tests will set s.PluginError if they are expecting an error
+	// back, and gotOnUpdateCatalogRequest will contain the value of
+	// the OnUpdateCatalogRequest for the current running test. Note
+	// that this means that the tests cannot run in parallel, but there
+	// could be other factors affecting that as well.
+	var gotOnUpdateCatalogRequest *plgpb.OnUpdateCatalogRequest
+	var pluginError error
+	testPlugin := hostplg.TestPlugin(t, dbConn, "test")
+	testPluginMap := map[string]plgpb.HostPluginServiceClient{
+		testPlugin.GetPublicId(): &WrappingPluginClient{
+			Server: &TestPluginServer{
+				OnUpdateCatalogFn: func(_ context.Context, req *plgpb.OnUpdateCatalogRequest) (*plgpb.OnUpdateCatalogResponse, error) {
+					gotOnUpdateCatalogRequest = req
+					return &plgpb.OnUpdateCatalogResponse{Persisted: &plgpb.HostCatalogPersisted{Secrets: req.GetNewCatalog().GetSecrets()}}, pluginError
+				},
+			},
+		},
+	}
+
+	// Set up two existing catalogs for duplicate tests, one with org
+	// scope, one with project scope.
```

Comment:
I think catalogs are only able to be project scoped.  At least that is enforced in the service handler.  It should probably be enforced deeper if that is something we actually want to enforce.

Reply:
Do we have an example of a similar constraint? I can work on adding it and a few tests in a new PR and then fix this up to just create/use two separate projects.

Reply:
I've just updated this to use two separate project scopes for now.